projects/storefrontlib/src/cms-components/checkout/components/shipping-address/shipping-address.component.ts

```diff
@@ -149,73 +118,46 @@ export class ShippingAddressComponent implements OnInit, OnDestroy {
       actions: [{ name: textShipToThisAddress, event: 'send' }],
       header: selected && selected.id === address.id ? textSelected : '',
     };
-
-    this.cards.push(card);
-
-    return card;
-  }
-
-  addressSelected(address: Address): void {
-    this.selectedAddress$.next(address);
   }
 
-  next(): void {
-    this.addAddress({ address: this.selectedAddress, newAddress: false });
+  selectAddress(address: Address): void {
+    this.checkoutDeliveryService.setDeliveryAddress(address);
   }
 
-  addAddress({
-    newAddress,
-    address,
-  }: {
-    newAddress: boolean;
-    address: Address;
-  }): void {
-    if (newAddress) {
-      this.checkoutDeliveryService.createAndSetAddress(address);
-      this.goTo = CheckoutStepType.DELIVERY_MODE;
-      return;
-    }
-    if (
-      this.setAddress &&
-      this.selectedAddress &&
-      this.setAddress.id === this.selectedAddress.id
-    ) {
-      this.goNext();
-    } else {
-      this.goTo = CheckoutStepType.DELIVERY_MODE;
-      this.checkoutDeliveryService.setDeliveryAddress(address);
-    }
-  }
-
-  addNewAddress(address: Address): void {
-    this.addAddress({ address, newAddress: true });
+  addAddress(address: Address): void {
+    this.userAddressService
+      .getAddresses()
+      .subscribe(addresses => {
+        if (addresses.includes(address)) {
+          this.checkoutDeliveryService.setDeliveryAddress(address);
+        } else {
+          this.checkoutDeliveryService.createAndSetAddress(address);
+        }
+        this.goNext();
+      })
+      .unsubscribe();
```

Comment:
Maybe it would be better to put this `unsubscribe` in the `OnDestroy` component lifecycle hook?